app/src/main/java/com/infomaniak/swisstransfer/ui/components/SwissTransferAlertDialog.kt

```diff
@@ -0,0 +1,134 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.swisstransfer.ui.components
+
+import androidx.annotation.StringRes
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.infomaniak.swisstransfer.R
+import com.infomaniak.swisstransfer.ui.theme.Margin
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SwissTransferAlertDialog(
+    modifier: Modifier = Modifier,
+    @StringRes titleRes: Int,
+    @StringRes descriptionRes: Int,
+    onDismiss: () -> Unit,
+    onConfirmation: () -> Unit,
+    shouldEnableConfirmButton: () -> Boolean = { true },
+    content: @Composable (ColumnScope.() -> Unit)? = null,
+) {
+    BasicAlertDialog(
+        onDismissRequest = onDismiss,
+        modifier = Modifier
+            .wrapContentHeight()
+            .fillMaxWidth(),
+    ) {
+        Card(shape = RoundedCornerShape(Margin.Medium)) {
+            BasicAlertDialogContent(
+                modifier = modifier,
+                titleRes = titleRes,
+                descriptionRes = descriptionRes,
+                additionalContent = content,
+                onDismiss = onDismiss,
+                onConfirmation = onConfirmation,
+                shouldEnableConfirmButton = shouldEnableConfirmButton,
+            )
+        }
+    }
+}
+
+@Composable
+private fun BasicAlertDialogContent(
+    modifier: Modifier,
+    @StringRes titleRes: Int,
+    @StringRes descriptionRes: Int,
+    additionalContent: @Composable (ColumnScope.() -> Unit)? = null,
+    onDismiss: () -> Unit,
+    onConfirmation: () -> Unit,
+    shouldEnableConfirmButton: () -> Boolean = { true },
+) {
+    Column(modifier.padding(Margin.Large)) {
+        TitleAndDescription(titleRes, descriptionRes)
+        additionalContent?.let {
+            it()
+            Spacer(Modifier.height(Margin.Large))
+        }
+        ActionButtons(onDismiss, onConfirmation, shouldEnableConfirmButton)
+    }
+}
+
+@Composable
+private fun TitleAndDescription(titleRes: Int, descriptionRes: Int) {
+    Text(
+        text = stringResource(titleRes),
+        style = SwissTransferTheme.typography.bodyMedium,
+        color = SwissTransferTheme.colors.primaryTextColor,
+    )
+    Spacer(Modifier.height(Margin.Large))
+    Text(
+        text = stringResource(descriptionRes),
+        style = SwissTransferTheme.typography.bodyRegular,
+        color = SwissTransferTheme.colors.secondaryTextColor,
+    )
+    Spacer(Modifier.height(Margin.Large))
```

Comment:
This Spacer is kinda hidden in here.

Shouldn't we move it to `BasicAlertDialogContent` ?

The concept of using `Spacer` instead of `Modifier.padding` is to more easily see the margins between components.
So I feel we should avoid putting Spacers inside components if they are on the edges of the component _(`top` and `bot` edges for Columns, `start` and `end` edges for Rows)_.

Reply:
I like the idea that a component takes care of everything to be exactly like we want it to be. And as there is no cases where we won't need a margin after the description, it feels right to me that the component has its spacer.
But I also understand we could want to see all the spacer in the "parent" component to separate its children